docker-shared.sh

```diff
@@ -71,13 +71,31 @@ push ()
 #   - build and push an image
 ci ()
 {
-  if [[ "$1" =~ (^(latest|staging-|production-|experimental))|(_cow$) && "$2" == "false" ]]
+  if [[ "$2" == "true" ]]
   then
+    echo "Skipping Docker image build due to pull-request status ($2)"
+  elif [[ ! "$1" =~ (^(latest|staging|production-|experimental))|(_cow$) ]]
+  then
+    echo "Skipping Docker image build due to uninteresting branch name ($1)"
+  elif [[ "$1" == "staging" ]]
```

Comment:
I think we can use the DockerHub registry API here to pull the `git.ref` label without having to pull the image. This will save us a lot of time, and given how fast the check will be, perhaps we can just perform this check for _all_ builds regardless of branch name.

The magic juju requires the `jq` binary (a CLI JSON parser) be available. I suspect the Travis images already have this installed, because it is super useful, it's like sed for json.

If you have jq, then all you need is this:

token=`curl -H 'Accept: application/json' --user "$DOCKERHUB_USER:$DOCKERHUB_PASSWORD" "https://auth.docker.io/token?service=registry.docker.io&scope=repository:rightscale/acl:pull" | jq --raw-output .token`

curl -H "Authorization: Bearer $token" "https://registry-1.docker.io/v2/rightscale/acl/manifests/latest" | jq --raw-output .history[0].v1Compatibility | jq --raw-output '.config.Labels["git.ref"]'